cmd/operator-sdk/generate/csv.go

```diff
@@ -171,6 +173,12 @@ Flags that change project default paths:
 				c.updateCRDs = false
 			}
 
+			// Check if the user has any specific preference to enable / disable interactive prompts.
+			// Default behaviour is to disable the prompts.
+			if !cmd.Flags().Changed("interactive") {
```

Comment:
It may be easier to detect explicit user action here:
```go
// Default is InteractiveSoftOff.
if cmd.Flags().Changed("interactive") {
	if c.interactive {
		c.interactivelevel = gencatalog.InteractiveHardOn
	} else {
		c.interactivelevel = gencatalog.InteractiveHardOff
	}
}
```

Then we don't have to check `c.interactive` elsewhere.

Reply:
`cmd.Flags().Changed()` needs to be inside `RunE:` field of cobra Command. And since we are assigning the value of `--interactive` flag to `c.interactive` outside the same, it is being checked twice.